### ISSUE TITLE
Deleting directory failed because 'storageclass.pathPattern' is defined

### ISSUE THREAD

**Issue by hxp-boy** [kubernetes-sigs/nfs-subdir-external-provisioner#272]
I guess it is because the default path is used when deleting the directory, and the path defined by `pathPattern`  is not used. Hope it can be resolved soon.
# nfs-subdir-external-provisioner   error log：
```bash
I0310 04:37:51.387493       1 controller.go:1450] delete "pvc-3877df78-3015-40f4-b659-c5396c25c3a9": started
W0310 04:37:51.392455       1 provisioner.go:146] path /persistentvolumes/nfs-pvc-test does not exist, deletion skipped
I0310 04:37:51.392481       1 controller.go:1478] delete "pvc-3877df78-3015-40f4-b659-c5396c25c3a9": volume deleted
I0310 04:37:51.400587       1 controller.go:1524] delete "pvc-3877df78-3015-40f4-b659-c5396c25c3a9": persistentvolume deleted
I0310 04:37:51.400606       1 controller.go:1526] delete "pvc-3877df78-3015-40f4-b659-c5396c25c3a9": succeeded
```
```bash
[root@k8s1 ~]# tree /nfs/
/nfs/
└── data
    └── minio-cluster
        └── nfs-pvc-test
```

# nfs-client-provisioner.yaml
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nfs-client-provisioner
  labels:
    app: nfs-client-provisioner
  # replace with namespace where provisioner is deployed
  namespace: nfs-provisioner
spec:
  replicas: 1
  strategy:
    type: Recreate
  selector:
    matchLabels:
      app: nfs-client-provisioner
  template:
    metadata:
      labels:
        app: nfs-client-provisioner
    spec:
      serviceAccountName: nfs-client-provisioner
      containers:
        - name: nfs-client-provisioner
          image: k8s.gcr.io/sig-storage/nfs-subdir-external-provisioner:v4.0.2
          imagePullPolicy: IfNotPresent
          volumeMounts:
            - name: nfs-client-root
              mountPath: /persistentvolumes
          env:
            - name: PROVISIONER_NAME
              value: k8s-sigs.io/nfs-subdir-external-provisioner
            - name: NFS_SERVER
              value: 192.168.0.228
            - name: NFS_PATH
              value: /nfs/data
      volumes:
        - name: nfs-client-root
          nfs:
            server: 192.168.0.228
            path: /nfs/data
```

# class.yaml
```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass 
metadata:
  name: nfs-provisioner-storage
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
provisioner: k8s-sigs.io/nfs-subdir-external-provisioner
parameters:
  onDelete: delete
  pathPattern: "${.PVC.namespace}/${.PVC.annotations.nfs.io/storage-path}"
```

# pvc.yaml
```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: nfs-pvc-test
  annotations:
    nfs.io/storage-path: "nfs-pvc-test"
spec:
  storageClassName: "nfs-provisioner-storage"
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 0.5Gi
```

**Comment by terrificsoysauce** [comment]
+1

Is [this](https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/commit/386af3ff8702b852d45d8bb86b25895b5aa2c69d#r110425625) the root cause?

**Comment by MidnessX** [comment]
> ```yaml
> # [...]
>       containers:
>         - name: nfs-client-provisioner
>           image: k8s.gcr.io/sig-storage/nfs-subdir-external-provisioner:v4.0.2
> # [...]
> ```
The problem is in the v.4.0.2 release, which is the latest available as a pre-built container image.

> +1
> 
> Is [this](https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/commit/386af3ff8702b852d45d8bb86b25895b5aa2c69d#r110425625) the root cause?

No, that's actually the fix, which is part of the ``v4.0.18`` release.
However, it looks like images aren't getting built anymore.
Issues #277 and #281 so far haven't received any answer.

The only real solution, at least for now, is to build an image yourself.
I've tested the following ``pathPattern`` with release ``v4.0.18`` and the NFS folder gets correctly deleted:
``` yaml
pathPattern: "${.PVC.namespace}/${.PVC.name}"
```

**Comment by xsoloking** [comment]
I built an image from master branch and tested on v1.27.3+k3s1, it still doesn't work. 

### pathPattern in Storage Class
 `pathPattern: "${.PVC.namespace}/${.PVC.annotations.nfs.io/storage-path}"`

### nfs-subdir-external-provisioner error log
PersistentVolumeClaim: test-222
```
event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"test-222", UID:"df63cfe0-ecd9-49d6-be17-2980af596f5e", APIVersion:"v1", ResourceVersion:"2742220", FieldPath:""}): type: 'Normal' reason: 'ProvisioningSucceeded' Successfully provisioned volume pvc-df63cfe0-ecd9-49d6-be17-2980af596f5e
I0810 09:18:09.006432       1 controller.go:1450] delete "pvc-df63cfe0-ecd9-49d6-be17-2980af596f5e": started
W0810 09:18:09.010285       1 provisioner.go:152] path /persistentvolumestest-222 does not exist, deletion skipped
```

I made a fix as fellow and built an arm64 image "soloking/nfs-subdir-external-provisioner:master",  it works for me now
### cmd/nfs-subdir-external-provisioner/provisioner.go
`oldPath := strings.Replace(path, p.path, mountPath + "/", 1)`
